akka-stream/src/main/scala/akka/stream/javadsl/Source.scala

```diff
@@ -569,16 +568,19 @@ final class Source[Out, Mat](delegate: scaladsl.Source[Out, Mat]) extends Graph[
    * Materializes this Source, immediately returning (1) its materialized value, and (2) a new Source
    * that can be used to consume elements from the newly materialized Source.
    *
-   * Prefer the method taking an ActorSystem unless you have special requirements.
+   * Note that the `ActorSystem` can be used as the `systemProvider` parameter.
    */
-  def preMaterialize(system: ActorSystem): Pair[Mat @uncheckedVariance, Source[Out @uncheckedVariance, NotUsed]] = {
-    val (mat, src) = delegate.preMaterialize()(SystemMaterializer(system).materializer)
+  def preMaterialize(systemProvider: ClassicActorSystemProvider)
+      : Pair[Mat @uncheckedVariance, Source[Out @uncheckedVariance, NotUsed]] = {
+    val (mat, src) = delegate.preMaterialize()(SystemMaterializer(systemProvider.classicSystem).materializer)
     Pair(mat, new Source(src))
   }
 
   /**
    * Materializes this Source, immediately returning (1) its materialized value, and (2) a new Source
    * that can be used to consume elements from the newly materialized Source.
+   *
+   * Prefer the method taking an `ActorSystem`/`ClassicActorSystemProvider` unless you have special requirements.
```

Comment:
I think we can write `ActorSystem` (only) here since that is what the parameter is in practice.

Reply:
Good point